### PR TITLE
Refactor Monitor API

### LINKED ISSUE #718: Refactor the APIs - Refactor the Monitor API and related implementations
Based on #660
- [ ] Refactor the Monitor API and related implementations in the backend-API.
- [ ] Update calling the Monitor API in the UI-client.
- [ ] Test Refactor Monitor API works well.

### PR DESCRIPTION
**Pull request by baixinsui**
fixed #718

### CODE REVIEW

File: modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java [hunk 1]
```diff
@@ -31,57 +29,66 @@
 import org.springframework.web.context.WebApplicationContext;
 
 @ExtendWith(SpringExtension.class)
-@ContextConfiguration(classes = {CredentialManageApi.class, CommonExceptionHandler.class,
-        CredentialCenter.class, MonitoringExceptionHandler.class, IdentityProviderManager.class})
+@ContextConfiguration(classes = {ServiceMetricsApi.class, MonitorMetricsService.class,
```
Comment (swaroopar) [inline modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java:32]: Should MonitorMetricsService class be also renamed to just MetricsService?

File: modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java [hunk 2]
```diff
@@ -31,57 +29,66 @@
...
+        MonitoringExceptionHandler.class, IdentityProviderManager.class})
 @WebMvcTest
 class MonitoringExceptionHandlerTests {
```
Comment (swaroopar) [inline modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java:35]: should MonitoringExceptionHandler and MonitoringExceptionHandlerTests be changed to ServiceMetricsExceptionHandler?

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added.

File: modules/api/src/main/java/org/eclipse/xpanse/api/ServiceMetricsApi.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    public ServiceMetricsApi(MetricsService metricsService) {
+        this.metricsService = metricsService;
+    }
+
+    /**
+     * Get metrics of a deployed service or a resource.
+     */
+    @Tag(name = "Monitor",
+            description = "APIs to get metrics of deployed services.")
```
Comment (iskey) [inline modules/api/src/main/java/org/eclipse/xpanse/api/ServiceMetricsApi.java:54]:
```suggestion
            description = "APIs for getting the metrics of deployed services.")
```

File: modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java (hunk 2 above)
Comment (baixinsui) [reply modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java]: Ok

File: modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java (hunk 1 above)
Comment (baixinsui) [reply modules/api/src/test/java/org/eclipse/xpanse/api/exceptions/handler/MonitoringExceptionHandlerTests.java]: Ok

File: modules/monitor/src/main/java/org/eclipse/xpanse/modules/monitor/MetricsService.java [hunk 4]
```diff
@@ -40,7 +40,7 @@
  */
 @Slf4j
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/monitor/src/main/java/org/eclipse/xpanse/modules/monitor/MetricsService.java:41]: @sonatype-lift ignore

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**